📊 food: Update food prices

pabloarosado opened the pull request:
Update data from World Bank's Food Prices for Nutrition dataset (to version 3.0).

NOTES:
* The main changes are in metadata, since several aspects of the methodology have changed since the previous version.
* I haven't approved charts in [Chart diff](http://staging-site-update-food-prices/etl/wizard/chart-diff) yet. I'll do that after confirming these things with Hannah.

paarriagadap commented on etl/steps/data/grapher/wb/2024-09-09/food_prices_for_nutrition.meta.yml:
```diff
@@ -0,0 +1,105 @@
+tables:
```
Why is this not done in garden?

pabloarosado replied:
When I created the step I thought it would be convenient to separate garden-related metadata from grapher-related metadata. Now we usually do all this in garden. I think it's fine either way.

paarriagadap commented on etl/steps/data/garden/wb/2024-09-09/food_prices_for_nutrition.meta.yml:
```diff
@@ -0,0 +1,325 @@
+definitions:
+  description_key_pct_of_population: &description_key_pct_of_population
```
2021 prices, where needed.

paarriagadap reviewed (approved):
Good! I included my comments below, mostly about metadata. Related to this, you should try to add to the subtitles `This data is adjusted for inflation and differences in the cost of living between countries.` whenever possible. I understand avery indicator here has this adjustment.

Also, where necessary, with monetary data you can add to the footnote `This data is expressed in [international-$](#dod:int_dollar_abbreviation) at 2021 prices.`

See more about this in the [harmonizing FASTT guide](https://docs.google.com/document/d/1tcknTryJAxOR0yv8x38s3jkONq-TY3T4HNCEcZQ_c-s/edit?pli=1#heading=h.xrsjsdraq83d)

pabloarosado commented on etl/steps/data/garden/wb/2024-09-09/food_prices_for_nutrition.py:
```diff
@@ -0,0 +1,70 @@
+    #
+    # Load meadow dataset and read its main table.
+    ds_meadow = paths.load_dataset("food_prices_for_nutrition")
+    tb = ds_meadow.read_table("food_prices_for_nutrition")
```
The method `read_table` was created as an optimized way to do `ds_meadow["..."].reset_index()`.

pabloarosado commented on etl/steps/data/garden/wb/2024-09-09/food_prices_for_nutrition.meta.yml:
```diff
@@ -0,0 +1,325 @@
+    - "The average share of non-food expenditure is based on the spending habits of specific income groups (quintiles):
+      - In upper-middle-income and high-income countries, the first quintile of consumers is considered (the poorest 20% of consumers), where people spend around 54% of their income on non-food needs.
+      - In low-income and lower-middle-income countries, the second quintile is considered (the second lowest 20% of consumers), where people spend about 37% and 44% of their income on non-food needs, respectively."
+    - "The international poverty lines (in international dollars based on 2017 prices) are:
```
I agree, but I adapted these definitions from [the World Bank metadata](https://databank.worldbank.org/source/food-prices-for-nutrition#selectedDimension_DBList) (click on the (i) button of, e.g. Series -> "Percent of the population who cannot afford sufficient calories"). Do you have any suggestion for how to call them otherwise?

paarriagadap replied:
For this dataset they call it like that, so I would keep it. I was just saying this as a side note. I don't think they have a collective name outside this dataset.

pabloarosado commented on etl/steps/data/garden/wb/2024-09-09/food_prices_for_nutrition.meta.yml:
```diff
@@ -0,0 +1,325 @@
+
+dataset:
+  update_period_days: 365
+  description: |
```
It's not completely necessary, but it's also not shown in data pages or the explorer (I think it's only available from the catalog?). I'll keep it for now.

pabloarosado commented on etl/steps/data/garden/wb/2024-09-09/food_prices_for_nutrition.meta.yml:
```diff
@@ -0,0 +1,325 @@
+          - *description_key_number_of_population
+      cost_of_a_healthy_diet:
+        title: Cost of a healthy diet
+        unit: current international-$ per person per day
```
That's what the World Bank's metadata say: "The cost of a healthy diet indicator is the cost of purchasing the least expensive locally available foods to meet requirements for energy and food-based dietary guidelines (FBDGs), in current PPP$/person/day, for a representative person within energy balance at 2330 kcal/day."
I suppose it would be best to adjust for inflation.

pabloarosado replied:
In my last commit, I have adjusted this column for inflation.

pabloarosado commented on etl/steps/data/garden/wb/2024-09-09/food_prices_for_nutrition.meta.yml:
```diff
@@ -0,0 +1,325 @@
+          - *description_nutrient_adequate_diet
+          - This indicator is calculated as the percentage of a country's population that is unable to afford a nutrient-adequate diet, multiplied by the country's population.
+          - *description_key_number_of_population
+      population:
```
This population indicator is not used in grapher, but I keep it in case there are some discrepancies with our population. This way it's easier to compare when debugging.

pabloarosado commented:
Thanks @paarriagadap for all your great suggestions!
In [my last commit](https://github.com/owid/etl/pull/3254/commits/ce11580265074deef20f4c76cebbadabe2cde23b) I have adjusted the cost of a healthy diet for inflation. I have checked [the World Bank's metadata](https://databank.worldbank.org/source/food-prices-for-nutrition#selectedDimension_DBList) of all cost variables, and that variable seems to be the only one given in current PPP dollars. All others are in constant 2021 PPP dollars, as far as I understand.
I have adjusted our metadata accordingly, and I'll adjust the subtitles and footnotes now.
Please have a quick look at the changes and let me know if you see something odd. Thanks!

paarriagadap commented on etl/steps/data/garden/wb/2024-09-09/food_prices_for_nutrition.py:
```diff
@@ -14,17 +14,42 @@
+# To express the healthy diet costs in constant 2021 PPP$, we need to adjust for inflation.
+# This is done by multiplying the cost in a given YEAR by the ratio CPI(BASE_YEAR) / CPI(YEAR).
+# Base year for CPI corrections.
+CPI_BASE_YEAR = 2021
```
The method looks good, but I am still not sure if it's 2021 or 2017 here. If the PPP rates used were in 2021 prices, this is fine. If they are in 2017 prices it should be 2017.

paarriagadap reviewed (approved):
Looks good! Though I am still not sure if these are 2021 or 2017 prices. See https://github.com/owid/owid-issues/issues/1622#issuecomment-2344401637

Marigold commented:
@pabloarosado those draft charts should be fixed on master and here.
